Makefile

```diff
@@ -0,0 +1,63 @@
+# Copyright 2022 The Kubernetes Authors.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Makefile for KRM functions registry.
+
+MYGOBIN = $(shell go env GOBIN)
+ifeq ($(MYGOBIN),)
+MYGOBIN = $(shell go env GOPATH)/bin
+endif
+export PATH := $(MYGOBIN):$(PATH)
+
+# Run all tests in this repo for in-tree and out-of-tree functions
+all: install-tools license verify-metadata test-in-tree
+
+## Run all unit tests and e2e tests for in-tree functions
+test-in-tree: e2e-test unit-test lint
+
+# Run all the e2e tests for in-tree functions
+e2e-test: build-local $(MYGOBIN)/kustomize
+	cd tests; \
+    go test -v ./...
+
+# Build all in-tree functions locally
+build-local:
+	cd krm-functions && $(MAKE) build-local
+
+
+# Add project licenses to all code files in here
+.PHONY: license
+license: $(MYGOBIN)/addlicense
+	( find . -type f -exec bash -c "$(MYGOBIN)/addlicense -y 2022 -c 'The Kubernetes Authors.' -f LICENSE_TEMPLATE {}" ";" )
+
+# Lint all in-tree functions
+lint:
+	cd tests; golangci-lint run ./...
+	cd krm-functions && find . -type f -name go.mod -execdir golangci-lint run ./... \;
+
+# TODO
+# Run all unit tests for in-tree functions
+unit-test:
+
+# TODO
+# Verify the metadata for all in-tree and out-of-tree functions
+verify-metadata:
+
+# Install tools needed to run tests
+install-tools: \
+	$(MYGOBIN)/kustomize \
+	$(MYGOBIN)/addlicense \
+	$(MYGOBIN)/golangci-lint
+
+# Install kustomize
+$(MYGOBIN)/kustomize:
```

Comment:
Why don't we use a static name for the target? e.g. `install-kustomize`

Reply:
This way, it only installs kustomize if it is not already there in the go bin. Similar things are in the [kustomize makefile](https://github.com/kubernetes-sigs/kustomize/blob/ff40460d3bc537a7c03b0ad9faa0987e26d40202/Makefile#L86). 

If we change the name of the target from this value to `install-kustomize`, then every time someone runs something like `make all` or `make test-in-tree`, it will refetch kustomize. To me this seems unnecessary and maybe even an undesirable side effect if the user does not want to upgrade to the latest version of kustomize. 

I would prefer to leave it as is, but one option if you feel strongly is to rename to `install-kustomize`, but as part of the make target, we can check if kustomize is already installed before trying to install it.

Reply:
IMO as long as kustomize is in PATH we don't need to install it again.
Can we do something like https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/Makefile#L105-L111? i.e. use the one in PATH if any, otherwise install it in GOBIN and use it.

Reply:
done! did this for install-kustomize, install-addlicense, and install-golangci-lint.